### PR TITLE
feat(inputtoggle): add an inputtoggle

### PR DESCRIPTION
**Pull request by mmmmmmaya**
### Summary
- **What:** Add `InputToggle` component
- **Ticket:** [[sc-157940]](https://app.shortcut.com/sci-design-system/story/157940)
- **Design:** https://www.figma.com/proto/EaRifXLFs54XTjO1Mlszkg/Science-Design-System-Reference?node-id=1876%3A21139&scaling=min-zoom&page-id=642%3A1

### Demos
after

### Checklist
- [x] I merged latest `main`
- [x] I manually verified the change
- [x] I added labels to my PR
- [x] I either asked design for a review or hid my changes behind a feature flag
- [x] I added relevant unit tests
- [x] Default Story in Storybook
- [x] LivePreview Story in Storybook
- [x] Test Story in Storybook
- [x] Variables from `defaultTheme.ts` used wherever possible
- [ ] If updating an existing component, depreciate flag has been used where necessary
- [ ] Chromatic build verified by @chanzuckerberg/sds-design

### CODE REVIEW

File: src/core/InputToggle/style.ts [hunk 1]
```diff
@@ -0,0 +1,154 @@
...
+  const corners = getCorners(props);
+  const spaces = getSpaces(props);
+  const TOGGLE_HEIGHT = 18;
+
+  return `
+    cursor: ${disabled ? "default" : "pointer"};
+    border-radius: ${corners?.l}px;
+    height: 26px;
+    width: 62px;
+    line-height: 18px;
```
Comment (j-x-han) [inline src/core/InputToggle/style.ts:29]: @chanzuckerberg/sds-design Are these values we'll reuse/should we define these as bases somewhere?

**Review by j-x-han — APPROVED** [review]
LGTM overall, just a question for design about some of the hardcoded values

File: src/core/InputToggle/style.ts (hunk 1 above)
Comment (Janeece) [reply src/core/InputToggle/style.ts:29]: Hmm I can imagine there are cases where we'd want the toggle to be able to increase the width if needed (like if we needed to use a longer word). We haven't really discussed that or how to handle hardcoded cases. 

Would we be able to allow products using this component to override the dimensions if needed?

**Review by Janeece — APPROVED** [review]
@mmmmmmaya I think this looks great! Not sure what the outcome will be from Julie's Question.

File: src/core/InputToggle/style.ts (hunk 1 above)
Comment (mmmmmmaya) [reply src/core/InputToggle/style.ts:29]: I believe comsumer could override with width, as written, by importing the sds toggle and then using emotion/styled.

**Pull request merged**